test/test_sentencepiece_ops.py

```diff
@@ -437,6 +476,33 @@ def test_string_sentencepiece_tokenizer_bin(self):
         for i in range(0, 2):
             assert_almost_equal(exp[i], py_txout[i])
             assert_almost_equal(exp[i], cc_txout[i])
+            
+    def test_xlm_roberta_tokenizer(self):
```

Comment:
in general, can we also add C++ unit tests to exercise new functionality? the reason I ask is that the Python tests don't run on certain platforms like Android and iOS.

Reply:
interesting - we should likely discuss this and revamp the entire testing process in another PR then i think? because we don't have C++ unit testing for any of the new functionality we have added recently (not to mention everything else, which we know works on Android/iOS because they're in prod but don't have unit testing for otherwise)

this file for example would need to include all the tokenizers, i only see wordpiece for now: https://github.com/microsoft/onnxruntime-extensions/blob/696e7502f16de207bad1f13d236af9455c6e26d6/test/static_test/test_tokenizer.cc

Reply:
if it is more involved to set up, another PR is fine. but it would be good to have more tests actually running on those platforms.

I'm currently working on enabling the C++ unit tests on iOS so hopefully we will have that soon.

Reply:
it requires a lot efforts and we also need know the value of this work.
Do we encounter any case that C++ code tests well with Python while it doesn't work on Android?
I know there is a lot of  if-else for mobile in build script, but there is every few #ifdef ANDROID in C++ code.

Reply:
> it requires a lot efforts and we also need know the value of this work.

The value is that we can be more confident about the code that we release on these platforms if there is more test coverage on the same platform.

> Do we encounter any case that C++ code tests well with Python while it doesn't work on Android?

Not aware of a Python test or equivalent failing on Android, but we don't run the Python tests on Android.
I will note that I encountered a C++ test failure that showed up on an Android device in the process of enabling those tests on Android.
The more testing we can do on Android and iOS the better.

FYI @skottmckay

Reply:
let's discuss in a separate thread since this is not specific to this PR - resolving this.